src/lib/BaseClient.ts

```diff
@@ -57,7 +57,11 @@ export abstract class BaseClient {
         let parsedResponse: any
         try {
           response = await this.rawFetch(path, init)
-          parsedResponse = await response.json()
+          if(response.status === 204) {
+            parsedResponse = {ok: true, data: null}
+          } else {
+            parsedResponse = await response.json()
```

Comment:
If you want the `BaseClient` to handle requests without responses, you can do that, generically, independently of the status code of the response, by checking if the body of the response is `null`. In those cases, you can set `parsedResponse` to `{ ok: true, data: null }` to let the flow continue and eventually return `null`.

Reply:
good catch